core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -613,14 +620,16 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
     setDaemon(true)
 
     private def monitorWorker(): Unit = {
-      // Kill the worker if it is interrupted, checking until task completion.
+      // Wait until the task is either canceled or completed.
       // TODO: This has a race condition if interruption occurs, as completed may still become true.
       while (!context.isInterrupted && !context.isCompleted) {
```

Comment:
Do we want `&& writerThread.isAlive` here ?

Reply:
Yeah, I think we might want that, though I'm having trouble reasoning about the effect it would have on worker reuse. We still want the monitor thread's lifetime to be tied to the Python worker's lifetime.

As I mentioned above, it seems like it would be easier to create a new kind of monitor thread for this case.